File: score-http/score-service/src/main/java/org/oagi/score/service/common/data/AppUser.java
package org.oagi.score.service.common.data;

import lombok.Data;

import java.math.BigInteger;

@Data
public class AppUser {

    private BigInteger appUserId;
    private String loginId;
    private String password;
    private String name;
    private String organization;
    private String email;
    private boolean emailVerified;
    private boolean developer;
    private boolean admin;
    private boolean enabled;

}

File: connect-center-api/app/services/models/app_user.py
"""Service models for app-user operations."""

from __future__ import annotations

from datetime import datetime
from typing import Literal

from dataclasses import dataclass

from app.types.identifiers import AppUserId


@dataclass(kw_only=True)
class AppUserServiceResult:
    """Application user information."""

    app_user_id: AppUserId
    login_id: str
    username: str
    organization: str | None
    email: str | None
    email_verified: bool
    email_verified_timestamp: datetime | None
    is_developer: bool
    is_admin: bool
    is_enabled: bool


@dataclass(kw_only=True)
class UserSummary:
    """Service-layer user summary."""

    user_id: AppUserId
    login_id: str
    username: str
    roles: list[Role]


Role = Literal["Admin", "Developer", "End-User"]
